### PR TITLE
Add a VCR module who intercepts, record and replay API requests

### PR DESCRIPTION
**Pull request by Aschen**
## Description

Add a VCR module inspired by https://github.com/linkedin/sepia.

When imported, this module will be activated by the presence of the `VCR_MODE` environment variable:
 - `record`: external call to API made with `fetch` will be executed and the the response saved to the filesystem
 - `playback`: responses to API calls will be read from the filesystem and an error will be thrown if the answer is missing

This module save the API request responses to the file system and identifies them by using a hash of the input parameters (url + verb + headers + body).

This PR also contains the recorded tapes for current tests so I also edited the CI to run tests using the `playback` mode.

fix https://github.com/huggingface/huggingface.js/issues/91 and https://github.com/huggingface/huggingface.js/issues/67

### CODE REVIEW

**Comment by Aschen** [comment]
@coyotte508 I still have problems with the requests made by the `commit.spec.ts` file, I will continue to investigate later

**Comment by coyotte508** [comment]
I'm a bit embarassed to say this after the PR is done, but the issue is on `@huggingface/inference`. There's no issue in `@huggingface/hub`.

> Also, I think it should be considered as a good practice to never run the tests again the real API in the CI.

The [python library](https://github.com/huggingface/huggingface_hub) and this library use a special backend on another environment: hub-ci.huggingface.co. The whole value of the libs is to interact with the Hugging Face hub, so we test against it - the python lib already caught some errors on the API that could've gone undetected.

**Comment by Aschen** [comment]
> I'm a bit embarassed to say this after the PR is done, but the issue is on @huggingface/inference. There's no issue in @huggingface/hub.

Ahah that kind of mistake! It's ok because the logic in the VCR module will works for the other package as well :+1: 

Technically if your hubs have a strong compliance with the no breaking change policy then it's not needed to execute the API requests but I understand if you can to double test it.

I will fix this PR later

**Comment by Aschen** [comment]
@coyotte508 I was working on this but I hit the rate limit before having opportunity to save the tapes. I will try again later, hopefully the limit will be reset

**Comment by coyotte508** [comment]
> @coyotte508 I was working on this but I hit the rate limit before having opportunity to save the tapes. I will try again later, hopefully the limit will be reset

Any token from any free token is good to have a much better rate limit. Also, no worries :)

**Comment by Aschen** [comment]
I generated a token to finish the recording of the tapes.

My main concern is the total size of those tapes because they are quite huge:
```
$ du -h test/tapes 
22M     test/tapes
```

File: .github/workflows/lint-and-test.yml [hunk 1]
```diff
@@ -49,6 +49,6 @@ jobs:
         run: pnpm --filter ...[${{ steps.since.outputs.SINCE }}] type-check
 
       - name: Test
-        run: pnpm --filter ...[${{ steps.since.outputs.SINCE }}] test:ci
+        run: VCR_MODE=playback pnpm --filter ...[${{ steps.since.outputs.SINCE }}] test:ci
```
Comment (coyotte508) [inline .github/workflows/lint-and-test.yml:52]: I know we have different opinions on this but it should only be playback when `HF_ACCESS_TOKEN` is not defined

**Review by coyotte508 — COMMENTED** [review]
+1,793,376 lines :grin: 

We should store tapes in LFS

**Comment by Aschen** [comment]
Apparently there is a limitation and I can't push the file to LFS because I'm working on a public fork https://github.com/git-lfs/git-lfs/issues/1906

Lets merge #107 and then I will try one of the workaround (removing and recreating my repo)

**Comment by Aschen** [comment]
@coyotte508 Unfortunately I couldn't push the tapes to LFS :(

What we can do:
  1) merge this PR and then open a new one with the tapes (you just need to run`VCR_MODE=record pnpm test` and commit the tapes)
  2) give me temporarily write access to the repo ("With great power comes great responsibility" would say uncle ben and so I)

**Pull request closed without merging**